internals/src/num/mod.rs

```diff
@@ -0,0 +1,23 @@
+//! Helpers for integers
+
+/// Extensions for signed integers
+pub trait IntExt {
+    /// the unsigned integer type
+    type Unsigned;
+
+    ///use this to find unsigned absolute if rust version less that 1.51
+    #[cfg(not(rust_v_1_51))]
+    fn unsigned_abs(self) -> Self::Unsigned;
+}
+
+impl IntExt for i64 {
```

Comment:
It might be useful to implement this for other types using a macro.

Reply:
Well, it's a private trait and we only use it once.

Reply:
I remember we had other uses, I don't remember which types. Maybe all of them are `i64`. :man_shrugging:

Reply:
How about we let @jim4067 work this in as a single imp and then I"ll macro-ize it once merged with all the other integer types? @jim4067 does that suit you?

Reply:
@tcharding, sounds good.